lib/charms/hpc_libs/v0/slurm_ops.py

```diff
@@ -257,52 +376,62 @@ def unset(self, *keys: str) -> None:
         _snap("unset", "slurm", *args)
 
 
-class MungeManager(ServiceManager):
-    """Manage `munged` service operations."""
+class _SnapMungekeyManager(MungekeyManager):
+    """Control the munge key using Snap."""
 
-    def __init__(self) -> None:
-        service = ServiceType.MUNGED
-        self._service = service
-        self.config = ConfigurationManager(service.config_name)
+    def _mungectl(self, *args: str, stdin: Optional[str] = None) -> str:
+        """Control munge via `slurm.mungectl ...`.
+
+        Args:
+            *args: Arguments to pass to `mungectl`.
+            stdin: Input to pass to `mungectl` via stdin.
+
+        Raises:
+            subprocess.CalledProcessError: Raised if `mungectl` command fails.
+        """
+        return _call("slurm.mungectl", *args, stdin=stdin)
 
-    def get_key(self) -> str:
+    def get(self) -> str:
         """Get the current munge key.
 
         Returns:
             The current munge key as a base64-encoded string.
         """
-        return _mungectl("key", "get")
+        return self._mungectl("key", "get")
 
-    def set_key(self, key: str) -> None:
+    def set(self, key: str) -> None:
         """Set a new munge key.
 
         Args:
             key: A new, base64-encoded munge key.
         """
-        _mungectl("key", "set", stdin=key)
+        self._mungectl("key", "set", stdin=key)
 
-    def generate_key(self) -> None:
+    def generate(self) -> None:
         """Generate a new, cryptographically secure munge key."""
-        _mungectl("key", "generate")
+        self._mungectl("key", "generate")
 
 
-class PrometheusExporterManager(ServiceManager):
-    """Manage `slurm-prometheus-exporter` service operations."""
-
-    def __init__(self) -> None:
-        self._service = ServiceType.PROMETHEUS_EXPORTER
+def _call(cmd: str, *args: str, stdin: Optional[str] = None) -> str:
+    """Call a command with logging.
 
+    Raises:
+        SlurmOpsError: Raised if the command fails.
+    """
+    cmd = [cmd, *args]
+    _logger.debug(f"Executing command {cmd}")
+    try:
+        return subprocess.check_output(cmd, input=stdin, stderr=subprocess.PIPE, text=True).strip()
+    except subprocess.CalledProcessError as e:
+        _logger.error(f"`{' '.join(cmd)}` failed")
+        _logger.error(f"stderr: {e.stderr.decode()}")
+        raise SlurmOpsError(f"command {cmd[0]} failed. Reason:\n{e.stderr.decode()}")
```

Comment:
[nit]: I wonder if we should just have the entire error message be lowercase? 

```suggestion
        raise SlurmOpsError(f"command {cmd[0]} failed. reason:\n{e.stderr.decode()}")
```